src/js/containers/bulkDownload/archive/AwardDataArchiveContainer.jsx

```diff
@@ -0,0 +1,189 @@
+        this.setAgencyList = this.setAgencyList.bind(this);
+        this.requestResults = this.requestResults.bind(this);
+        this.updateFilter = this.updateFilter.bind(this);
+    }
+
+    componentDidMount() {
+        this.setAgencyList();
+        this.requestResults();
+    }
+
+    setAgencyList() {
+        this.setState({
+            inFlight: true
+        });
+
+        if (this.agencyListRequest) {
+            this.agencyListRequest.cancel();
+        }
+
+        // perform the API request
+        this.agencyListRequest = BulkDownloadHelper.requestAgenciesList({
+            agency: 0
+        });
+
+        this.agencyListRequest.promise
+            .then((res) => {
+                const cfoAgencies = res.data.agencies.cfo_agencies;
+                const otherAgencies = res.data.agencies.other_agencies;
+                this.setState({
+                    agencies: {
+                        cfoAgencies,
+                        otherAgencies
+                    }
+                });
+            })
+            .catch((err) => {
+                console.log(err);
+                this.agencyListRequest = null;
+            });
+    }
+
+    updateFilter(name, value) {
+        const filters = Object.assign({}, this.state.filters, {
+            [name]: value
+        });
+
+        this.setState({
+            filters
+        });
+    }
+
+    requestResults() {
+        this.setState({
+            inFlight: true
+        });
+
+        if (this.resultsRequest) {
+            this.resultsRequest.cancel();
+        }
+
+        // perform the API request
+        if (this.state.filters.id === 'all') {
+            // don't include the agency parameter to request all agencies
+            this.resultsRequest = BulkDownloadHelper.requestArchiveFiles({
+                fiscal_year: parseInt(this.state.filters.fy, 10),
+                type: this.state.filters.type.name
+            });
+        }
+        else {
+            this.resultsRequest = BulkDownloadHelper.requestArchiveFiles({
+                agency: parseInt(this.state.filters.agency.id, 10),
+                fiscal_year: parseInt(this.state.filters.fy, 10),
+                type: this.state.filters.type.name
+            });
+        }
+
+        this.agencyListRequest.promise
+            .then((res) => {
+                this.parseResults(res.data.monthly_files);
+            })
+            .catch((err) => {
+                console.log(err);
+                this.resultsRequest = null;
+            });
+    }
+
+    parseResults(data) {
+        const results = [];
+
+        data.forEach((item) => {
+            // Format Agency String
+            let formattedAgency = item.agency_name;
+            if (item.agency_acronym) {
+                formattedAgency = `${item.agency_name} (${item.agency_acronym})`;
+            }
+
+            // Format Updated Date
+            const date = moment(item.updated_date);
```

Comment:
Can you specify the parse input format? This helps avoid ambiguities (ie, what is the month in 2017-04-03) https://momentjs.com/docs/#/parsing/string-format/

Reply:
Confirmed the API is returning `YYYY-MM-DD`